Kudu.Core/Jobs/BaseJobRunner.cs

```diff
@@ -140,6 +185,10 @@ private void CacheJobBinaries(JobBase job, IJobLogger logger)
                     UpdateAppConfigs(tempJobInstancePath);
 
                     _workingDirectory = tempJobInstancePath;
+
+                    // cache the file map snapshot for next time (to aid in detecting
+                    // file deletions)
+                    _cachedSourceDirectoryFileMap = sourceDirectoryFileMap;
```

Comment:
On first call of CacheJobBinaries, WorkingDirectory will be null, so we don't do the JobDirectoryHasChanged check at all. We copy/cache the job files and set WorkingDirectory at that point, **along with _cachedSourceDirectoryFileMap**. So on subsequent runs that will be set and the map contents will be the initial unmodified set of files we cached.

Is my understanding correct here? I'm going to do some e2e testing now to confirm.